scripts/genpinctrl/genpinctrl.py

```diff
@@ -410,8 +410,13 @@ def get_mcu_signals(data_path, gpio_ip_afs):
             if pin.get("Type") != "I/O":
                 continue
 
-            # obtain pin port (A, B, ...) and number (0, 1, ...)
             pin_name = pin.get("Name")
+
+            # skip duplicate remappable entries in some G0 files
+            if family == "STM32G0" and pin_name in ("PA9", "PA10"):
+                continue
```

Comment:
I understand that for such specific conditions, ad hoc code saves us from great complexity.
But thinking about potential upcoming series support, is there a way to generate a warning when duplicate nodes are seen ?
Not a blocker as I understand that zephyr dtlib will report issues anyway.

Reply:
> Not a blocker as I understand that zephyr dtlib will report issues anyway.

I'll merge with your approval, then, thank you. I would like to get the fix in for v3.2.

Reply:
> I understand that for such specific conditions, ad hoc code saves us from great complexity. But thinking about potential upcoming series support, is there a way to generate a warning when duplicate nodes are seen ? Not a blocker as I understand that zephyr dtlib will report issues anyway.

They are not strictly duplicate, but rather redundant entries because G0 has the "single" and "remappable" entries, which basically encode the same information. And it's not consistent with F0 series (where pin remap functionality is also available). It would be good to clarify why XML is structured like that before doing any changes.

Reply:
> It would be good to clarify why XML is structured like that before doing any changes.

Ok, I'll investigate